numpy/core/src/multiarray/numpyos.c

```diff
@@ -451,6 +458,17 @@ NumPyOS_ascii_strtod(const char *s, char** endptr)
     struct lconv *locale_data = localeconv();
     const char *decimal_point = locale_data->decimal_point;
     size_t decimal_point_len = strlen(decimal_point);
+    /*
+     * It is unclear why this function needs locale information.
+     * The goal is to do locale-independent parsing. This function
+     * contains code to stop the parsing early if an unusual decimal
+     * point is encountered and then pass the string so far to
+     * NumPyOS_ascii_strtod_plain, which will also stop if an unusual
+     * decimal point is encountered. The only case I can think of
+     * where this would matter is if the decimal separator were
+     * something that could occur in the middle of a valid number,
+     * in which case this function does the wrong thing.
```

Comment:
Under Python 2.6, PyOS_ascii_strtod refuses to parse the string at all, and returns an error instead of "1" which was the correct best-efforts result.